### PR TITLE
Improve translation of LATERAL FLATTEN

### PR DESCRIPTION
**Pull request by vil1**

### CODE REVIEW

File: core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4 [hunk 1]
```diff
@@ -3158,9 +3158,13 @@ predicatePartial
...
     | NOT? IN L_PAREN (subquery | exprList) R_PAREN
     | NOT? likeExpression
-    | NOT? BETWEEN expr AND expr
+    | NOT? BETWEEN temporalExpr AND temporalExpr
     ;
 
+temporalExpr: functionCall | primitiveExpression
+
+	;
+
```
Comment (nfx) [inline core/src/main/antlr4/com/databricks/labs/remorph/parsers/snowflake/SnowflakeParser.g4:3167]:
```suggestion
temporalExpr
    : functionCall 
    | primitiveExpression
	;

```

blank lines are eye sores

File: core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/plans.scala [hunk 2]
```diff
@@ -133,6 +133,24 @@ abstract class Plan[PlanType <: Plan[PlanType]] extends TreeNode[PlanType] {
       case other => Nil
     }.toSeq
   }
+
+  /** Returns all expressions present in this query plan and its descendants */
+  def nestedExpressions: Seq[Expression] = {
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/plans.scala:138]: why do we need to modify a widely-used class with the method that's used only once?

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/FlattenLateralViewToExplode.scala [hunk 3]
```diff
@@ -5,26 +5,61 @@ import com.databricks.labs.remorph.parsers.snowflake.NamedArgumentExpression
...
-      // val path = named.get("PATH").orElse(Some(Literal("")))
-      val outer = getFlag(named, "OUTER")
-      // val recursive = getFlag(named, "RECURSIVE")
-      // val mode = named.get("MODE").orElse(Some(Literal("BOTH")))
-      val explode = Explode(input)
-      Lateral(TableFunction(explode), outer = outer)
+  private val FLATTEN_OUTPUT_COLUMNS = Set("seq", "key", "path", "index", "value", "this")
+
+  override def apply(plan: LogicalPlan): LogicalPlan = {
+
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/FlattenLateralViewToExplode.scala:11]:
```suggestion
```
i wonder why scalafmt doesn't pick it up

File: core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/FlattenLateralViewToExplode.scala [hunk 4]
```diff
@@ -5,26 +5,61 @@ import com.databricks.labs.remorph.parsers.snowflake.NamedArgumentExpression
...
+      id -> FLATTEN_OUTPUT_COLUMNS.filter { col =>
+        exprs.exists(_.find {
+          case Dot(x, Id(c, false)) => x == id && c.equalsIgnoreCase(col)
+          case Column(Some(r), Id(c, false)) => r.head == id && c.equalsIgnoreCase(col)
+          case _ => false
+        }.nonEmpty)
+      }
+    }.toMap
+
+    val translatedToExplode = plan transform {
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/FlattenLateralViewToExplode.scala:30]: fun one, right?

**Review by nfx — CHANGES_REQUESTED** [review]

**Comment by nfx** [comment]
@vil1 also fix py failures

**Comment by vil1** [comment]
Will be done by @aman-db in another PR

**Pull request closed without merging**